File: leetcode/lc-us/cpp/2303.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    double calculateTax(vector<vector<int>> &brackets, int income) {
        double res = 0;
        double prev = 0;
        for (auto &b: brackets) {
            if (income >= b[0]) {
                res += (b[0] - prev) * b[1] / 100.0;
            } else {
                res += (income - prev) * b[1] / 100.0;
                break;
            }
            prev = b[0];
        }
        return res;
    }
};

File: leetcode/lc-us/py/2303.py
from typing import List


class Solution:
    def calculateTax(self, brackets: List[List[int]], income: int) -> float:
        res = 0
        pre = 0
        for bracket in brackets:
            upper, percent = bracket
            if upper <= income:
                res += (upper - pre) * percent / 100
            else:
                res += (income - pre) * percent / 100
                break
            pre = upper
        return res
